src/nwb2bids/_converters/_run_config.py

```diff
@@ -0,0 +1,81 @@
+import pathlib
+import typing
+
+import pydantic
+
+from .._core._file_mode import _determine_file_mode
+from .._core._home import _get_home_directory
+from .._core._run_id import _generate_run_id
+from .._core._validate_existing_bids import _validate_existing_directory_as_bids
+
+
+class RunConfig(pydantic.BaseModel):
+    """
+    Specifies configuration options for a single run of NWB to BIDS conversion.
+
+    bids_directory : directory path
+        The path to the directory where the BIDS dataset will be created.
+        Defaults to the current working directory and checks if it is either empty or a BIDS dataset.
+    additional_metadata_file_path : file path, optional
+        The path to a YAML file containing additional metadata not included within the NWB files
+        that you wish to include in the BIDS dataset.
+    file_mode : one of "move", "copy", or "symlink"
+        Specifies how to handle the NWB files when converting to BIDS format.
+            - "move": Move the files to the BIDS directory.
+            - "copy": Copy the files to the BIDS directory.
```

Comment:
note mentioning somewhere that it would be CoW if filesystem supports and Python 3.13+.